### ISSUE TITLE
Search for translation sources with custom extensions via make-json

### ISSUE THREAD

**Issue by olegglocc** [wp-cli/i18n-command#231]
`make-json` extract JavaScript strings from PO files with .js and .min.js files extensions while generating individual JSON files like it should. But I have JS files with other extensions (for example .tag), which ignored this way.

In [/i18n-command/blob/master/src/MakeJsonCommand.php](https://github.com/wp-cli/i18n-command/blob/master/src/MakeJsonCommand.php) start at 134 line I found code which responsible for it. So there inside iteration it check for each extension and return source if was found. Off course, I can "hardcode" and this will work. Something like:
```
if ( substr( $file, - 4 ) === '.tag' ) {
    return $file;
}
```
But my request is in core update from your side, because I think a lot of people will use with feature for their own purposes and there can be a great variety of similar extensions. Feature must be multipurpose for any extension and, desirable, provide this opportunity to programmers at their own discretion (like `--extensions=tag` argument or something similar). I.e additional to .js and .min.js will search for extensions specified in `--extensions` argument.

Or maybe you have any other suggestions for resolving my request? I am open to discussion and ready to listen to any suggestions.

Thanks!

**Comment by olegglocc** [comment]
Hi!

Have you got some updates about this issue?

**Comment by swissspidy** [comment]
No uodate I‘m afraid. I personally haven‘t had time to look into it but it sounds like a feasible enhancement and PRs are more than welcome.

You do have some exotic file extensions I must say :-)

**Comment by michaelw85** [comment]
@olegglocc @swissspidy I've created a [PR](https://github.com/wp-cli/i18n-command/pull/439) to add this feature.

I was wondering, how do you end up with a PO file containing custom JS extensions? 
Is this already supported by the create pot command?

**Comment by swissspidy** [comment]
> Is this already supported by the create pot command?

The `wp i18n make-pot` command only supports `.js` and `.jsx` files.

But they might be using something else to extract the strings.

**Comment by michaelw85** [comment]
> > Is this already supported by the create pot command?
> 
> The `wp i18n make-pot` command only supports `.js` and `.jsx` files.
> 
> But they might be using something else to extract the strings.

Yes, I was also thinking something along those lines. I think we should add a new ticket to add support to the make pot although I don't know how complex it is to add this feature.
